### ISSUE TITLE
Ignore XML comment lines

### ISSUE THREAD

**Issue by UlyssesZh** [markdownlint/markdownlint#571]
## Describe the Enhancement:

Ignore a line if it is a comment. A test case may look like this:

````md
<!-- MD041 is not triggered -->
# Test block comments
<!-- MD022 is not triggered -->

- So this list is still regarded as wrapped by blank lines
- goes on
<!-- Even if the comment is very long, it does not trigger MD013 because it is ignored -->

<!-- A block comment can contain
multiple lines
like this -->
```plaintext
same with code blocks, MD031 is not triggered
<!-- Block comments in code blocks are not ignored, so a lone line like this can still trigger MD013 -->
```

Inline<!-- comments --> are not ignored, so a long line like this can also trigger MD013

<!-- Even if this comment is ignored, the two line endings do not trigger MD012 -->

<!-- or MD047 -->
````

### Impacted Rules:

MD012, MD013, MD022, MD031, MD041, MD047.

## Describe the Need:

A feature parity of the Node.js implementation. This is also necessary to implement the feature of temporarily setting rules like `<!-- markdownlint-disable MD013 -->`.

## Current Alternative

No alternatives.

## Can We Help You Implement This?:

At first my idea is to remove the lines covered by elements such that `e.type == :xml_comment && e.options[:category] == :block` and then parse the text with removed lines again. However, this poses some problem: (1) rules like MD012 and MD047 still need to be applied to the original doc, not the doc with removed comments; (2) XML comments immediately following a block-type element are not parsed as a comments block by kramdown but as a comments span; (3) auto-fix should be applied to the original doc instead of the doc with removed lines. My next idea then is to pick out all comments using a regexp and then remove the lines covered by them except those inside code blocks, but this approach may be hard to cover all edge cases. I need to hear what you think about this before working on a PR.
